handlers.go

```diff
@@ -0,0 +1,167 @@
+package firecracker
+
+import (
+	"context"
+	"fmt"
+)
+
+// Handler name constants
+const (
+	StartVMMHandlerName                = "StartVMM"
+	BootstrapLoggingHandlerName        = "BootstrapLogging"
+	CreateMachineHandlerName           = "CreateMachine"
+	CreateBootSourceHandlerName        = "CreateBootSource"
+	AttachDrivesHandlerName            = "AttachDrives"
+	CreateNetworkInterfacesHandlerName = "CreateNetworkInterfaces"
+	AddVsocksHandlerName               = "AddVsocks"
+)
+
+// StartVMMNamedHandler is a named handler that will
+// handle starting of the VMM. This handler will also
+// set the exit channel on completion.
+var StartVMMNamedHandler = NamedHandler{
```

Comment:
What kind of problem are you trying to solve with this? Why not to call functions directly?

Reply:
Makes testing easier when testing individual components also allows for extendable functionality without the `Init` function becoming this huge behemoth. Allows the machine to not require any knowledge of what it is executing as well, making it a little more of a black box of commands. All the machine should care about is whether or not a command has failed.

Reply:
Why not use interfaces instead?

Reply:
I think I'm comfortable with the handler approach.  It does add a lot of flexibility; the main concern that I have right now is whether we need it.  But I also don't really object to having it.